packages/contracts/contracts/Unirep.sol

```diff
@@ -125,6 +127,14 @@ contract Unirep is IUnirep, zkSNARKHelper, VerifySignature {
             revert AttesterIdNotMatch(attesterId);
     }
 
+    function verifyNullifier(uint256 nullifier) private {
+        if (nullifier == 0) return;
```

Comment:
I don't think we should have this here. We should instead conditionally call `verifyNullifier`.

Reply:
In the `spendReputation` function?

Reply:
Yes, because we don't need the non-0 safety check for epk nullifiers.